src/odemis/acq/test/fastem_conf_test.py

```diff
@@ -0,0 +1,113 @@
+# -*- coding: utf-8 -*-
+"""
+Created on 03 December 2021
+
+@author: Sabrina Rossberger, Éric Piel
+
+Copyright © 2021 Sabrina Rossberger, Delmic
+
+This file is part of Odemis.
+
+Odemis is free software: you can redistribute it and/or modify it under the terms
+of the GNU General Public License version 2 as published by the Free Software
+Foundation.
+
+Odemis is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+Odemis. If not, see http://www.gnu.org/licenses/.
+
+"""
+from __future__ import division
+
+import logging
+import math
+import os
+import unittest
+
+import odemis
+from odemis import model
+from odemis.acq import fastem_conf
+from odemis.acq.fastem_conf import OVERVIEW_MODE, LIVESTREAM_MODE, MEGAFIELD_MODE
+from odemis.util import img
+
+logging.getLogger().setLevel(logging.DEBUG)
+logging.basicConfig(format="%(asctime)s  %(levelname)-7s %(module)s:%(lineno)d %(message)s")
+
+# * TEST_NOHW = 1: not connected to anything => skip most of the tests
+# * TEST_NOHW = 0: connected to the real hardware or simulator
+# technolution_asm_simulator/simulator2/run_the_simulator.py
+TEST_NOHW = os.environ.get("TEST_NOHW", "0")  # Default is HW/simulator testing
+
+CONFIG_PATH = os.path.dirname(odemis.__file__) + "/../../install/linux/usr/share/odemis/"
+FASTEM_CONFIG = CONFIG_PATH + "sim/fastem-sim.odm.yaml"
+
+
+class TestFASTEMConfig(unittest.TestCase):
+    """Test the FASTEM scanner configuration."""
+
+    @classmethod
+    def setUpClass(cls):
+        if TEST_NOHW is True:
+            raise unittest.SkipTest("No simulator running or HW present. Skip fastem acquisition tests.")
+
+        cls.scanner = model.getComponent(role="e-beam")
+        cls.sed = model.getComponent(role="se-detector")
+        cls.multibeam = model.getComponent(role="multibeam")
+        cls.mppc = model.getComponent(role="mppc")
+
+    def test_configure_scanner(self):
```

Comment:
It might be nice to split this up into 3 test cases. Now if the overview mode test fails the rest doesn't get tested. While if you have a test case for each mode, everything is always tested.